File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/infra/PokemonProperty.java
package com.mdgd.pokemon.ui.pokemon.infra;

public interface PokemonProperty {
    int PROPERTY_IMAGE = 1;
    int PROPERTY_LABEL = 2;
    int PROPERTY_TITLE = 3;
    int PROPERTY_LABEL_IMAGE = 4;

    int getType();

    default int getNestingLevel() {
        return 0;
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/pokemon/infra/PokemonProperty.kt
package com.mdgd.pokemon.ui.pokemon.infra

interface PokemonProperty {
    val type: Int
    val nestingLevel: Int
        get() = 0

    companion object {
        const val PROPERTY_IMAGE = 1
        const val PROPERTY_LABEL = 2
        const val PROPERTY_TITLE = 3
        const val PROPERTY_TEXT = 4
    }
}
